Functions/Mock.Tests.ps1

```diff
@@ -1343,3 +1379,82 @@ Describe 'Mocking New-Object' {
         Assert-MockCalled New-Object
     }
 }
+
+Describe 'Mocking a function taking input from pipeline' {
+    $psobj = New-Object -TypeName psobject -Property @{'PipeIntProp'='1';'PipeArrayProp'=1;'PipeStringProp'=1}
+    $psArrayobj = New-Object -TypeName psobject -Property @{'PipeArrayProp'=@(1)}
+    $noMockArrayResult = @(1,2) | PipelineInputFunction
+    $noMockIntResult = 1 | PipelineInputFunction
+    $noMockStringResult = '1' | PipelineInputFunction
+    $noMockResultByProperty = $psobj | PipelineInputFunction -PipeStr 'val'
+    $noMockArrayResultByProperty = $psArrayobj | PipelineInputFunction -PipeStr 'val'
+
+    Mock PipelineInputFunction { write-output 'mocked' } -ParameterFilter { $PipeStr -eq 'blah' }
+
+    context 'when calling original function with an array' {
+        $result = @(1,2) | PipelineInputFunction
+        it 'Returns actual implementation' {
+            $result[0].keys | % {
+                $result[0][$_] | Should Be $noMockArrayResult[0][$_]
```

Comment:
While these loops do result in fewer lines of code, the output of a failed test is less useful (since we'd be seeing $_ in the logs).  Would it be worth it to just explicitly check each property of your function's hashtable?  Lots of duplication, but much clearer failure messages.